lib/wor/authentication/sessions_controller.rb

```diff
@@ -2,59 +2,78 @@ module Wor
         end
       end
 
       def invalidate_all
-        # should we rescue anything here ?
-        # if invalidating uses db and fails, or something like that
-        entity_custom_validation_invalidate_all_value(current_entity)
+        update_custom_validation_value(current_entity)
+        update_renew_id(current_entity)
         head :ok
       end
 
-      def generate_access_token(entity)
-        renew_id = token_renew_id
-        payload = entity_payload(entity).merge(
-          entity_custom_validation: entity_custom_validation_value(entity),
-          expiration_date: new_token_expiration_date,
-          maximum_useful_date: token_maximum_useful_date,
-          renew_id: renew_id
-        )
-        access_token_object(token_key, payload, renew_id)
+      ##########################################################################################
+      #                   DEFAULT METHOD IMPLEMENTATIONS, USER SHOULD CHANGE                   #
+      ##########################################################################################
+      def authenticate_entity(params)
+        entity = User.find_by(email: authentication_params[:email])
+        return nil unless entity.present? && entity.valid_password?(authentication_params[:password])
+        entity
       end
 
-      def renew_access_token(entity)
-        payload = decoded_token.payload
-        payload[:expiration_date] = new_token_expiration_date
-        payload[:entity_custom_validation] = entity_custom_validation_renew_value(entity)
-        Wor::Authentication::TokenManager.new(token_key).encode(payload)
+      def entity_payload(entity)
+        { id: entity.id }
       end
 
-      private
+      def new_token_expiration_date
+        Wor::Authentication.expiration_minutes.minutes.from_now.to_i
+      end
 
-      def access_token_object(token_key, payload, renew_id)
-        {
-          token: Wor::Authentication::TokenManager.new(token_key).encode(payload),
-          renew_id: renew_id
-        }
+      def new_token_renew_date
+        Wor::Authentication.renew_minutes.minutes.from_now.to_i
+      end
+
+      def token_renew_id(entity)
+        if entity.respond_to?(:renew_id)
+          current_key = entity.renew_id
+          return current_key if current_key
+          update_renew_id(entity)
+        end
+      end
+
+      def update_custom_validation_value(entity)
+        SecureRandom.hex(32).tap do |random_value|
+          begin
+            entity.update!(entity_custom_validation: random_value)
+          rescue
+            Rails.logger.info('User does not have a entity_custom_validation attribute')
+          end
+        end
       end
 
+      def update_renew_id(entity)
+        SecureRandom.hex(32).tap do |random_value|
+          begin
+            entity.update!(renew_id: random_value)
+          rescue
+            Rails.logger.info('User does not have a renew_id attribute')
```

Comment:
if the entity is invalid it would also raise an Exception. Maybe you want to rescue a specific Exception ?

Reply:
The idea behind this was to identify if the user had the attribute and was able to update it, we could probably rescue something more specific